cli/src/main/kotlin/commands/ClearlyDefinedUploadCommand.kt

```diff
@@ -113,11 +139,35 @@ class ClearlyDefinedUploadCommand : CliktCommand(
     override fun run() {
         val absoluteInputFile = inputFile.normalize()
         val curations = absoluteInputFile.readValue<List<PackageCuration>>()
-        val service = ClearlyDefinedService.create(server, OkHttpClientHelper.buildClient())
 
         var error = false
 
-        curations.forEachIndexed { index, curation ->
+        val curationsToCoordinates = curations.associateWith { it.id.toClearlyDefinedCoordinates().toString() }
+        val definitions = getDefinitions(curationsToCoordinates.values)
+
+        val uploadableCurations = curations.filter { curation ->
+            val harvestStatus = definitions[curationsToCoordinates[curation]]?.getHarvestStatus()
+
+            print("Package '${curation.id.toCoordinates()}' ")
+            when (harvestStatus) {
+                HarvestStatus.HARVESTED -> {
+                    println("has been harvested, will try to upload curations.")
+                    true
+                }
+
+                HarvestStatus.NOT_HARVESTED, HarvestStatus.PARTIALLY_HARVESTED -> {
+                    println("has not been harvested yet, cannot upload curations.")
+                    false
+                }
+
+                else -> {
```

Comment:
Could this be `null` instead of `else` to be more specific?

Reply:
Yes, fixed.